### DISCUSSION TITLE
Watermark Remove?

### DISCUSSION THREAD

**Discussion by naddel91** [roleoroleo/yi-hack-Allwinner-v2#940] [General]
Is there an option to remove the "YI" logo watermark on the bottom left?

**Comment by roleoroleo** [comment]
You can change it with another image but you can't remove it.
I don't remember esactly the name of the file:
```
root@yi-hack:/home/app# ls -la *.bmp
-rw-rw-r--    1 root     root         27704 Nov  6  2020 main.bmp
-rwxrwxr-x    1 root     root         27704 Nov  6  2020 main_kami.bmp
-rw-rw-r--    1 root     root          3128 Nov  6  2020 sub.bmp
-rwxrwxr-x    1 root     root          3128 Nov  6  2020 sub_kami.bmp
```
main_* is for high res and sub_* for low_res
If you want to change it with another file, you should use mount bind command, because the /home partition is read-only.
And you must comply with the format of the bmp file.
Another detail, alpha channel is not supported.

**Comment by luckyfreddy** [comment]
I've tested it. In my case, it was not possible to fully remove the watermark or get full transparency - but it is much less annoying.

1st.) the bitmap-file must be at least 16x16 pixel.
I've tried files with lower resolutions, but then the camera does not boot successfully and reboots continuously. 
I have used the following file, which i attached here "shadow_watermark.bmp" (it is inside zip-file, because this site does not allow to upload bitmap files [shadow_watermark_bitmap.zip](https://github.com/user-attachments/files/18800030/shadow_watermark_bitmap.zip) it works for me. 

2nd) I uploaded the bitmap to my camera into `/tmp/sd/yi-hack`

3rd) In the file `/tmp/sd/yi-hack/script/system.sh` I pasted below the first line (the `#!/bin/sh`, wich is called shebang) the following lines:

```
mount --bind /tmp/sd/yi-hack/shadow_watermark.bmp /home/app/main.bmp
mount --bind /tmp/sd/yi-hack/shadow_watermark.bmp /home/app/main_kami.bmp
mount --bind /tmp/sd/yi-hack/shadow_watermark.bmp /home/app/sub.bmp
mount --bind /tmp/sd/yi-hack/shadow_watermark.bmp /home/app/sub_kami.bmp
```

4th) reboot

After rebooting there is only a little half-transparency square into the left edge:
![watermark-screenshot](https://github.com/user-attachments/assets/16a73ada-3f76-4b09-b29e-a7cb4d4a00d4)

Please note:
If something goes wrong (maybe the bitmap file is broken or whatever) the rtsp-service does not start up. If that happens, the internal watchdog reboots the camera. After rebooting you've got a minute to log in with ssh and remove the added lines (from step 3). After that the camera will boot normal. It is necessary that the mount-commands stays at the top under the shebang in the
`/tmp/sd/yi-hack/script/system.sh` file - otherwise the services (rtsp) already started up with the watermark from the firmware. Please be aware, that the modifications from the file `/tmp/sd/yi-hack/script/system.sh` probably will only last as long, as an firmware upgrade is installed. After upgradeing the firmware you have to add the lines from step 3 again.

**Comment by Trevsweb** [comment]
would be cool if this was an option to replace the images/use the minimal mode in the UI

**Reply by luckyfreddy** [reply]
> would be cool if this was an option to replace the images/use the minimal mode in the UI

I use the workaround ( https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/940#discussioncomment-12201287 ) since February 14, 2025 without any problems at 5 Yi Home Cameras (y211ga) and would appreciate it, to. 

Maybe this could be an option for future releases, @roleoroleo ?

**Reply by roleoroleo** [reply]
Yes, it could be...
